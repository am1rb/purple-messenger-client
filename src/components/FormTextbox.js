// @flow

import React from 'react'
import classNames from 'classnames'

// @flow

import FormBaseInput from './FormBaseInput'

import styles from './FormTextbox.css'

type Props = {
	value: string,
	onChange: Function,
};


export default class FormTextbox extends React.PureComponent<Props>
{
	render()
	{
		return (
			<FormBaseInput
				{...this.props}
				renderInput={(className, props) => {
					return (
						<textarea className={className} {...props} />
					)	
				}}
			/>
		)
	}
}