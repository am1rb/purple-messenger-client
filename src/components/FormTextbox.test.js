// @flow

import React from 'react'
import renderer from 'react-test-renderer'

import FormTextbox from './FormTextbox'


describe("<FormTextbox /> component tests", () => {

	it("Should not regress", () => {

		const component = renderer.create(
			<FormTextbox
				value="This is sample text"
				onChange={() => {}}
			/>
		)

		expect(component.toJSON()).toMatchSnapshot()

	})

})