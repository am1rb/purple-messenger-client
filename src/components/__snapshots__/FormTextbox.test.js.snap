// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormTextbox /> component tests Should not regress 1`] = `
<textarea
  className="undefined"
  onChange={[Function]}
  value="This is sample text"
/>
`;
